server/oauth-core/src/main/java/org/eclipse/lyo/server/oauth/core/utils/AbstractAdapterCredentialsFilter.java

```diff
@@ -288,85 +298,110 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
 			boolean isTwoLeggedOAuthRequest = false;
 			String twoLeggedOAuthConsumerKey = null;
 
+            log.trace("CredentialsFilter - RequestURL: {} {} on session {}", request.getMethod(), request.getRequestURL().toString(), request.getSession().getId());
+
 			//TODO: possibly do this in web.xml
 			if (isProtectedResource(request))
 			{
+                log.trace("{} on session {} - A protected resource", request.getPathInfo(), request.getSession().getId());
 
 				// First check if this is an OAuth request.
 				try {
 					try {
 						OAuthMessage message = OAuthServlet.getMessage(request, null);
+		                log.trace("{} on session {} - checking for oauth1 authentication", request.getPathInfo(), request.getSession().getId());
 						// test if this is a valid two-legged oauth request
 						if ("".equals(message.getToken())) {
+	                        log.trace("{} on session {} - an oauth1 token with empty-value is found. checking for TwoLeggedOAuth", request.getPathInfo(), request.getSession().getId());
 							validateTwoLeggedOAuthMessage(message);
+			                log.trace("{} on session {} - It is a TwoLeggedOAuthRequest", request.getPathInfo(), request.getSession().getId());
 							isTwoLeggedOAuthRequest = true;
 							twoLeggedOAuthConsumerKey = message.getConsumerKey();
 						}
 
 						if (!isTwoLeggedOAuthRequest && message.getToken() != null) {
+                            log.trace("{} on session {} - an oauth1 token is found. Validating it.", request.getPathInfo(), request.getSession().getId());
 							OAuthRequest oAuthRequest = new OAuthRequest(request);
 							oAuthRequest.validate();
+                            log.trace("{} on session {} - an oauth1 token is valid", request.getPathInfo(), request.getSession().getId());
 							Connection connector = tokenToConnectionCache.get(message.getToken());
 							if (connector == null) {
+	                            log.trace("{} on session {} - an oauth1 token is valid, but no Connector is associated with it. Raising an exception TOKEN_REJECTED", request.getPathInfo(), request.getSession().getId());
 								throw new OAuthProblemException(
 										OAuth.Problems.TOKEN_REJECTED);
 							}
 
+                            log.trace("{} on session {} - oauth1 authentication is valid. Done. Binding the Connector to the session", request.getPathInfo(), request.getSession().getId());
 							request.getSession().setAttribute(CONNECTOR_ATTRIBUTE, connector);
 						}
 					} catch (OAuthProblemException e) {
+			            log.error("{} on session {} - OAuthProblemException caught. {}", request.getPathInfo(), e.toString());
```

Comment:
Same question for all `.error` log statements here as above. I think a `.warn()` may be better. In particular, what's the differnce between `OAuthProblemException` and `OAuthException`?

Reply:
fixed.
Note that some of these exceptions are raised within this same function. For example, see line 330 where the exception is raised.

Reply:
What I am trying to see is if we can cleanly split oauth exception handling into two catch statements: one catches "bad" signatures aka user input and another one catches oauth library internal errors. That's why I asked about the difference between the two exceptions.